### PR TITLE
2FA Modal Promise Should Reject With Error

### PR DESCRIPTION
**Pull request by mattlockyer**
This PR fixes the 2FA modal that was previously reporting success and not throwing errors properly.

It will now reject in the modal promise with the errors from contract helperand for no code / incorrect code and also reject with error if the user cancels the modal.

@Patrick1904 you may want to dive in and customize the error messaging in the UI.

See code changes for where/how errors are being thrown and potential spots to customize by throwing your own WalletError.

### CODE REVIEW

**Comment by mattlockyer** [comment]
@Patrick1904 suggested @vgrichina review now and he will update messaging async

**Comment by Patrick1904** [comment]
> This PR fixes the 2FA modal that was previously reporting success and not throwing errors properly.
> 
> It will now reject in the modal promise with the errors from contract helperand for no code / incorrect code and also reject with error if the user cancels the modal.
> 
> @Patrick1904 you may want to dive in and customize the error messaging in the UI.
> 
> See code changes for where/how errors are being thrown and potential spots to customize by throwing your own WalletError.

Great I will update messaging in follow-up PR.

File: src/utils/wallet.js [hunk 1]
```diff
@@ -111,6 +111,8 @@ class Wallet {
         this.accountId = localStorage.getItem(KEY_ACTIVE_ACCOUNT_ID) || ''
 
         this.twoFactor = new TwoFactor(this)
+
+        console.log(this)
```
Comment (vgrichina) [inline src/utils/wallet.js:115]: debug leftover?

File: src/components/accounts/two_factor/TwoFactorVerifyModal.js [hunk 2]
```diff
@@ -76,7 +77,7 @@ const TwoFactorVerifyModal = ({ open, onClose }) => {
         <Modal
             id='two-factor-verify-modal'
             isOpen={open}
-            onClose={() => onClose(false)}
+            onClose={() => onClose(false, 'The action has been cancelled')}
```
Comment (vgrichina) [inline src/components/accounts/two_factor/TwoFactorVerifyModal.js:80]: why pass a string to `onClose` vs `WalletError`?

**Comment by mattlockyer** [comment]
@Patrick1904 commented the relevant lines for messages / translation. Please have a look at the comment additions in https://github.com/near/near-wallet/pull/826/commits/b5831d5d1ca5d505591285d852b83a79259e358b

**Comment by vgrichina** [comment]
@kcole16 may you double check this works as expected, as I'm a bit deep in other stuff now.

**Comment by kcole16** [comment]
@mattlockyer Closing the modal seems to work now (receiving an error message, and able to resubmit).

Getting some very weird behavior when adding 2FA to an account though. 
- Create an account
- Initiate adding either email or phone 2FA
- Try entering the verification code

With both email and phone, the verification code seems to fail (no error though), and then I keep receiving verification codes. 

Eventually worked, but took 3 tries. Tried clearing cache as well, same situation.

Not happening on testnet.

**Comment by mattlockyer** [comment]
> Eventually worked, but took 3 tries. Tried clearing cache as well, same situation.
> 
> Not happening on testnet.

Looking into this. If it's timeouts they've been happening in these bursts where nothing works for 5-10 minutes and then it's back.

**Comment by kcole16** [comment]
Approving since I can’t reproduce now, so could’ve just been timeout related. We should keep an eye out for this in mainnet testing.

**Comment by Patrick1904** [comment]
I'm not able to reproduce the bug reported by @kcole16 either. Approving.

**Pull request merged**